### ISSUE TITLE
Bug: Post Saving Notice Continues — Even After the Post Is Notified to Be Saved

### ISSUE THREAD

**Issue by ahmadawais** [WordPress/gutenberg#5784]
After the CloudFlare bug was addressed I have again started using Gutenberg in production.

On a post bigger than 5,000 words, when I update my post, it shows the post is "Saving…" and then notifies that it is saved whereas the "Saving…" notice is still there.

I made a quick GIF.

![image](http://on.ahmda.ws/qQxv/c)

**Comment by markjaquith** [comment]
I could not duplicate this.

I tried with 5,000 and 10,000 character posts.

I tried with a fast connection, and a slow (100Kbps with 1000ms of latency) connection.

For me, the green bar (from a previous update) goes away as soon as a new update is started. When the new update ends, the "Saving" indicator goes away, the "Updating..." button turns to "Update", and the green "post updated" bar returns.

Tested with ffd9d53e20a32f5434990dca3e8f3e10a5ca5bcb

**Comment by ahmadawais** [comment]
@markjaquith Here's how you can replicate the conditions.

- All the text was in the TinyMCE text block (migrated posts)
- My site is behind CloudFlare free CDN/Firewall
- I can provide the system report with list of plugins installed
- Hosted on Kinsta.com

Now I can confirm it not only happens with large amount of text but any post even one with 42 words. 

---

### Take a look 👇

![](http://on.ahmda.ws/qRML/c)

---

### Here're the network stats.

![image](http://on.ahmda.ws/qSOA/c)

**Comment by markjaquith** [comment]
Aha. I figured it out. It's because of classic metaboxes. I installed a dummy one and now I see what you're seeing.

The green "Post updated!" bar appears after the `posts` request comes back, but the "Updating..." and "Saving" indicators do not reset until the subsequent metabox request comes back.

![kapture 2018-03-28 at 10 09 19](https://user-images.githubusercontent.com/353790/38034473-5bbdb144-3270-11e8-9953-e5b87022d23f.gif)

Perhaps we should delay the green "Post updated!" bar until that second request returns? That would be the safest. We might be able to have "Updating..." and "Saving" reset after the first request comes back, but I don't know if that might leave the UI in a dangerous state, where you can hit "Update" again, but it's not ready to do another update because it's waiting for the metaboxes to refresh.

**Comment by ahmadawais** [comment]
Exactly. It should definitely be fixed. A simple observer that subscribes to all the events in the lifecycle of an update/publish request would do 👍

**Comment by ahmadawais** [comment]
@danielbachhuber I can confirm that the bug is still present.

![image](https://on.ahmda.ws/rirb/c)

Gutenberg Version 2.9.0

**Comment by danielbachhuber** [comment]
Thanks @ahmadawais, I've filed it as a bug.

**Comment by youknowriad** [comment]
This was a duplicate of #5868 and is now fixed.

**Comment by amberhinds** [comment]
Has this been fixed in WordPress core? We're seeing the "post updated" message before save of post meta has completed (button still reads "updating...") and if editors click the view post link in the success message, it's interrupting the saving of post meta.
